test/versioned/mongodb/collection-common.js

```diff
@@ -273,30 +272,25 @@ function populate(db, collection) {
  * Bootstrap a running MongoDB instance by dropping all the collections used
  * by tests.
  * @param {*} mongodb MongoDB module to execute commands on.
- * @param {*} collections Collections to drop for test.
+ * @param {Array} collections Collections to drop for test.
  */
-function dropTestCollections(mongodb, collections) {
-  return common.connect(mongodb).then((res) => {
-    const client = res.client
-    const db = res.db
-
-    async.eachSeries(
-      collections,
-      (collection, cb) => {
-        db.dropCollection(collection, (err) => {
-          // It's ok if the collection didn't exist before
-          if (err && err.errmsg === 'ns not found') {
-            err = null
-          }
-
-          cb(err)
-        })
-      },
-      (err) => {
-        return common.close(client, db).catch((err2) => {
-          throw err || err2
-        })
+async function dropTestCollections(mongodb, collections) {
+  if (!collections.length) {
+    return
+  }
+
+  const { client, db } = await common.connect(mongodb)
+
+  const dropCollectionPromises = collections.map(async (collection) => {
+    try {
+      await db.dropCollection(collection)
+    } catch (err) {
+      if (err && err.errmsg !== 'ns not found') {
+        throw err
       }
-    )
+    }
   })
+
+  await Promise.all(dropCollectionPromises)
```

Comment:
If one of the `dropCollection` calls throw out of the catch, the `await` here will unpack and throw the error. In that situation, the `common.close()` will not attempt to execute.

So while messier, I think we'll need a try-finally to ensure the close gets called.